internal/plumbing/uast/uast.go

```diff
@@ -14,9 +14,9 @@ import (
 	"sync"
 	"time"
 
+	"github.com/Jeffail/tunny"
 	"github.com/gogo/protobuf/proto"
-	"github.com/jeffail/tunny"
-	"gopkg.in/bblfsh/client-go.v3"
+	bblfsh "gopkg.in/bblfsh/client-go.v3"
```

Comment:
This seems to be not required, the package name is already `bblfsh`
```suggestion
	"gopkg.in/bblfsh/client-go.v3"
```

Reply:
@vmarkovtsev it's better to follow imports generated by [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports). This tool is integrated into most IDEs and editors and makes sure the format of imports is the same, similar to gofmt.

If I run it on the current master:
```
$ goimports -d internal/plumbing/uast/uast.go
diff -u internal/plumbing/uast/uast.go.orig internal/plumbing/uast/uast.go
--- internal/plumbing/uast/uast.go.orig	2019-03-06 18:58:46.000000000 +0100
+++ internal/plumbing/uast/uast.go	2019-03-06 18:58:46.000000000 +0100
@@ -16,7 +16,7 @@

 	"github.com/gogo/protobuf/proto"
 	"github.com/jeffail/tunny"
-	"gopkg.in/bblfsh/client-go.v3"
+	bblfsh "gopkg.in/bblfsh/client-go.v3"
 	"gopkg.in/bblfsh/sdk.v2/uast/nodes"
 	"gopkg.in/bblfsh/sdk.v2/uast/nodes/nodesproto"
 	"gopkg.in/src-d/go-git.v4"
```

Reply:
I see. But why does it change that import?

Actually, GoLand highlights the additional name as a warning.

![image](https://user-images.githubusercontent.com/2793551/53904195-cb586600-4045-11e9-944c-1c5bcd1333df.png)

I mean, I am not really happy to comply with the bugs of a third-party tool, even if it is official and widespread. Tomorrow it may decide to change other imports while reverting this one.

Reply:
> I see. But why does it change that import?
>
> Actually, GoLand highlights the additional name as a warning.

GoLand may use an old version of this tool underneath. It's a quite recent change. You may get the opposite warning when you update GoLand. Read more:
https://github.com/golang/go/issues/29556
https://github.com/golang/go/issues/28428
https://go-review.googlesource.com/c/tools/+/152000

> I mean, I am not really happy to comply with the bugs of a third-party tool, even if it is official and widespread. Tomorrow it may decide to change other imports while reverting this one.

tools do change behaviors sometimes. For example, `gofmt` output was changed in go1.11:
https://golang.org/doc/go1.11#gofmt

It's definitely up to you as you are a maintainer of this project but go community in general tries to use official instruments for formatting. I'm pretty sure if you expect external contributions, most of them will include this change.

Reply:
Even more, it looks like GoLand uses goimports by default: https://www.jetbrains.com/help/go/integration-with-go-tools.html (I don't use this IDE, so I can't say for sure, sorry)
So most probably you have different behaviour because of outdated goimports in your GOPATH.

Reply:
OK, thanks for the links, now I see that it is a feature, not a bug. Quite funny that Russ Cox was confused with that himself. And I love that the change was made to run the tool "faster", although it is already blazing fast, and humble users have to pay for this additional +1% speed gain and change the code everywhere.

Aaaah. Good old Go way. Hate everywhere, every time.

Reply:
tbh I fully support this change. It's very hard to understand where the import comes from if the url has a different name from package name as in this case.

Reply:
According to the original commit message in https://go-review.googlesource.com/c/tools/+/152000 nobody cared about readability actually. So the benefit you mention is a surprising positive side effect.

Reply:
interesting discussion, I had never noticed the discrepancy previously 😅